pkg/sql/offsets_adapter.go

```diff
@@ -2,18 +2,22 @@ package sql
 
 type OffsetsAdapter interface {
 	// AckMessageQuery the SQL query and arguments that will mark a message as read for a given consumer group.
-	AckMessageQuery(topic string, row Row, consumerGroup string) (string, []interface{})
+	AckMessageQuery(topic string, row Row, consumerGroup string) Query
 
 	// ConsumedMessageQuery will return the SQL query and arguments which be executed after consuming message,
 	// but before ack.
 	//
 	// ConsumedMessageQuery is optional, and will be not executed if query is empty.
-	ConsumedMessageQuery(topic string, row Row, consumerGroup string, consumerULID []byte) (string, []interface{})
+	ConsumedMessageQuery(topic string, row Row, consumerGroup string, consumerULID []byte) Query
 
 	// NextOffsetQuery returns the SQL query and arguments which should return offset of next message to consume.
-	NextOffsetQuery(topic, consumerGroup string) (string, []interface{})
+	NextOffsetQuery(topic, consumerGroup string) Query
 
 	// SchemaInitializingQueries returns SQL queries which will make sure (CREATE IF NOT EXISTS)
 	// that the appropriate tables exist to write messages to the given topic.
-	SchemaInitializingQueries(topic string) []string
+	SchemaInitializingQueries(topic string) []Query
+
+	// BeforeSubscribingQueries returns queries which will be executed before subscribing to a topic.
+	// All queries will be executed in a single transaction.
+	BeforeSubscribingQueries(topic string, consumerGroup string) []Query
```

Comment:
Does it make sense to pass `ctx` here or allow returning an error?

Reply:
I was thinking about that, but I don't see use case for that (I even started refactoring rest of methods to add context). It would make sense if it would execute query, but in that case it's just generating it.